holoviews/core/ndmapping.py

```diff
@@ -169,11 +170,7 @@ def _add_item(self, dim_vals, data, sort=True, update=True):
 
         for dim, val in valid_vals:
             vals = self._cached_index_values[dim.name]
-            if vals == 'initial': self._cached_index_values[dim.name] = []
-            if not self._instantiated and self.get_dimension(dim).values == 'initial':
-                if val not in vals:
-                    self._cached_index_values[dim.name].append(val)
-            elif vals and val is not None and val not in vals:
```

Comment:
This was my early hacky approach to define a fixed order, I think it has been broken for a long time.